errors/error.go

```diff
@@ -270,14 +301,47 @@ func (e *Error) error(verbose bool) string {
 	return strings.Join(errParts, separator)
 }
 
+func (e *Error) defaultErrorParts() []interface{} {
+	return []interface{}{
+		e.FileRange,
+		e.Kind,
+		e.Description,
+		e.Stack,
+		e.Err,
+	}
+}
+
 // Error returns the error message.
 func (e *Error) Error() string {
-	return e.error(false)
+	return e.error(e.defaultErrorParts(), false)
 }
 
 // Detailed returns a detailed error message.
 func (e *Error) Detailed() string {
-	return e.error(true)
+	return e.error(e.defaultErrorParts(), true)
+}
+
+// AsList returns the error as a list.
+// If it's underlying error is a *List, then it just returns it because
+// they're already explicitly wrapped.
+func (e *Error) AsList() *List {
+	if errs, ok := e.Err.(*List); ok {
+		return errs
+	}
+
+	return L(e)
+}
+
+// Message returns the error message without some metadata.
+// This method is suitable for editor extensions that needs to handle the
+// metadata themselves.
+func (e *Error) Message() string {
```

Comment:
yep I feel the same. The `Error()` method I kept with the filerange because this is probably the best default, like if you only print the error then it have everything. But it's also not so easy to customize the "printer" of the error...
if you have an idea let me know.

Reply:
Hmm you already have a reasonably nice internal API to personalize which information will be present on the final message and in which order, but not sure we need that exposed :thinking:. All good for now, don't have better ideas yet =P.